protobuf/avs.proto

```diff
@@ -40,17 +40,57 @@ message SyncTasksReq {
 }
 
 enum TriggerType {
-  TimeTrigger          = 0;
-  ContractQueryTrigger = 1;
-  ExpressionTrigger    = 2;
+  ManualTrigger        = 0;
+  FixedEpochTrigger  = 1;
+  CronTrigger        = 2;
+  BlockTrigger       = 3;
+  EventTrigger         = 4;
+}
+
+message FixedEpochCondition {
+  repeated int64 epoches = 1;
+}
+// Simple timebase or cron syntax.
+message CronCondition {
+  repeated string cron_table = 1;
+}
+
+message BlockCondition {
+  int64 interval = 1;
+}
+
+// An arbitrary expression to express the condition.
+// People can define condition example
+//   chainlinkPrice("address-of-eth-usd-pair") > 2644500 && queryContract("contractaddress", "callmsg")[2] < = 5
+// By allow arbitrary expression, people can mix and match to create conplex
+// condition that match their workload
+//
+// The function to be used need to be pre-defined on our task egnine runtime.
+// When a new block is build, our engine will execute these check
+//
+// The expression language is re-present by https://expr-lang.org/
+message EventCondition {
+  string expression = 1;
 }
 
 message TaskTrigger {
   TriggerType            trigger_type   = 1;
 
-  TimeCondition          schedule       = 2;
-  ContractQueryCondition contract_query = 3;
-  ExpressionCondition    expression     = 4;
+  oneof trigger_condition {
+    bool                   manual     = 2;
+
+    // run at a specific epoch, name inspired by unix `at` utility
+    FixedEpochCondition    at      = 3;
+
+    // interval such as every hour/day/ etc can be converted to cronsyntax by the sdk/studio
+    CronCondition          cron       = 4;
+   
+    // currently the only support syntax is every <x> blocks
+    BlockCondition         block      = 5;
+
+    // support filter by event expression such as topic0, topic1, topoc2 and event_data and contract_address
+    EventCondition         event      = 6;
+  }
```

Comment:
I noticed that trigger_condition is combined across multiple `trigger_type` values. For `TimeTrigger` we will need sub-category such as `Fixed` and`Cron`, however, for `BlockTrigger`, we don’t actually need an additional `trigger_condition` field. Even we do, the trigger_condition should not re-use the same definition as the one for `TimeTrigger`.
This ties into the same design question I raised this morning for the UI.

Current return value in example.js
```
trigger: {
  trigger_type: 'BlockTrigger',
  block: { interval: '5' },
  trigger_condition: 'block'
},
```